pkg/rtc/wrappedreceiver.go

```diff
@@ -47,10 +55,139 @@ func (r *WrappedReceiver) DetermineReceiver(codec webrtc.RTPCodecCapability) {
 	}
 }
 
-func (r *WrappedReceiver) Codecs() []webrtc.RTPCodecCapability {
-	codecs := make([]webrtc.RTPCodecCapability, 0, len(r.receivers))
-	for _, receiver := range r.receivers {
-		codecs = append(codecs, receiver.Codec().RTPCodecCapability)
-	}
+func (r *WrappedReceiver) Codecs() []webrtc.RTPCodecParameters {
+	codecs := make([]webrtc.RTPCodecParameters, len(r.receivers))
+	copy(codecs, r.codecs)
 	return codecs
 }
+
+type DumbReceiver struct {
```

Comment:
What is the purpose of `DumbReceiver`? Seems like it is keeping state. So, am curious.

Reply:
For a backup codec, we don't have a webrtcReceiver before it published, so create a DumbReceiver to take position of the WebRtcReceiver to negotiate with subscriber downtrack, then we know the actual codec that subscriber needed, feed back to the publisher (through SubscribedCodecQuality), after publisher publish the needed backup codec, DumpReceiver upgrade to webrtcReceiver, everything work as normal